src/testsets.jl

```diff
@@ -141,6 +146,11 @@ end
 
 Get the properties associated with tests within a testset. Can be extended for custom
 testsets. Will return `nothing` for testsets which do not support test properties.
+
+When generating a JUnit XML report these will be the properties associated with all
+`testcase` elements contained within a `testsuite` element.
```

Comment:
So they will be duplicated?

Reply:
No. The properties created through `record_testset_property` are associated to the testset containing the call and all testsets nested within that testset. The properties created through `record_test_property` are associated with the tests contained within the testset containing the function call including all nested tests. When rendering the JUnit report this means that calls to `record_testset_property` are associated with `<testsuite>` elements while calls to `record_test_property` are associated with `<testcase>` elements.

For example:
```julia
using Test, TestReports, EzXML

ts = @testset ReportingTestSet "Root" begin
    record_testset_property("foo", 1)
    record_test_property("bar", 2)

    @testset "Inner" begin
        @test 1 == 1
    end

    @test 2 != 1
end;

prettyprint(report(ts))
```
Results in:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="2" failures="0" errors="0">
  <testsuite name="Root" tests="1" failures="0" errors="0" time="0.080" timestamp="2024-05-02T09:00:58.607" hostname="awro" id="0">
    <testcase name="2 != 1" id="1" classname="Root" time="0.000">
      <properties>
        <property name="bar" value="2"/>
      </properties>
    </testcase>
    <properties>
      <property name="foo" value="1"/>
    </properties>
  </testsuite>
  <testsuite name="Root/Inner" tests="1" failures="0" errors="0" time="0.053" timestamp="2024-05-02T09:00:58.634" hostname="awro" id="1">
    <testcase name="1 == 1" id="1" classname="Root/Inner" time="0.011">
      <properties>
        <property name="bar" value="2"/>
      </properties>
    </testcase>
    <properties>
      <property name="foo" value="1"/>
    </properties>
  </testsuite>
</testsuites>
```

Reply:
Got it. this clarified things a lot.
Can this be added into the docs somewhere?
Either in the docstring or in the manual?

Reply:
I've updated the examples in the docstring for `record_testset_properties` to include some of these details. I ended up not including the raw XML output as I thought it was getting a bit too verbose. Hopefully, the example is as clear as this comment. Let me know what you think